Sort player names in overlay, option to hide own name, show pid in debug

conker-rsc opened the pull request:
Modified the player name overlay behavior to:

1. No longer display your own name
2. Maintain a fixed sort-order of names to prevent the "dancing" that occurs when multiple players occupy the same tile

Hubcapp commented:
Don't believe you can see your own PID anymore, which is an oversight. Maybe acceptable to just add it to the debug menu, next to username.

Hubcapp commented:
maybe also just needs to be an option to display your own player name :s

Hubcapp commented:
need a toggle on that & also would like a keybind

Hubcapp reviewed (changes requested):
Requesting that this receive both a toggle & a keybind & also as a feature request maybe put PID in the debug menu after your player username.

conker-rsc commented:
Updated as requested

Hubcapp commented:
dislike the behaviour that ctrl-p can no longer toggle own player overhead name. Would like to replace this segment:

```java
            } else if (Settings.SHOW_PLAYER_NAME_OVERLAY.get(Settings.currentProfile)
                && npc.name != null
                && Client.player_name != null
                && !npc.name.equals(Client.player_name)) {
              showName = true;
            } else if (Settings.SHOW_OWN_NAME_OVERLAY.get(Settings.currentProfile)
                && npc.name != null
                && npc.name.equals(Client.player_name)) {
              showName = true;
            }
```

with

```java
            } else if (Settings.SHOW_PLAYER_NAME_OVERLAY.get(Settings.currentProfile) {
              boolean isOwnName = npc.name != null && npc.name.equals(Client.player_name);
              if (isOwnName) {
                showName = Settings.SHOW_OWN_NAME_OVERLAY.get(Settings.currentProfile);
              } else {
                showName = true;
              }
            }
```

Sorry but just feels very unnatural to have the settings not be linked at all & it's cumbersome to have to use the unintuitive alt-v keyboard shortcut in addition to ctrl-p to toggle both features at the same time (current behaviour), since they don't even share any common keys.

If SHOW_OWN_NAME should be possible without showing other players names, would like that as yet another setting SHOW_OWN_NAME_ALONE

conker-rsc commented:
No prob, I wasn't sure exactly how you wanted it to behave, per the last request. I've updated the logic to consider the hierarchies of settings here, and updated the settings label as well.

Hubcapp commented:
it is OK but the verbiage when toggling Show Own Name via keyboard shortcut might be better if it were updated for the new behaviour
